Public/Src/Engine/Scheduler/Artifacts/FileContentManager.cs

```diff
@@ -2267,13 +2269,25 @@ private static PipOutputOrigin GetPipOutputOrigin(ContentMaterializationOrigin o
                         if (!isAvailable)
                         {
                             Possible<ContentDiscoveryResult, Failure>? existingContent = null;
-
                             bool isPreservedOutputFile = IsPreservedOutputFile(pipInfo.UnderlyingPip, materializingOutputs, fileArtifact);
+
                             bool shouldDiscoverContentOnDisk =
                                 Configuration.Schedule.ReuseOutputsOnDisk ||
                                 !Configuration.Schedule.StoreOutputsToCache ||
                                 isPreservedOutputFile;
 
+                            var assumeContentUpToDate = isPreservedOutputFile &&
+                               Configuration.Distribution.BuildRole == DistributedBuildRoles.None
+                                && m_preservedUpToDateCacheLookupFiles.Contains(fileArtifact);
+
+                            if (materializingOutputs && state != null && assumeContentUpToDate)

```

Comment:
>assumeContentUpToDate [](start = 73, length = 21)

Can you put some comment why we need to consider `assumeContentUpToDate` here?

Reply:
i decided to remove that optimization.